modules/budget/main.tf

```diff
@@ -46,6 +46,29 @@ resource "google_billing_budget" "budget" {
     credit_types_treatment = var.credit_types_treatment
     services               = local.services
     labels                 = var.labels
+    calendar_period        = var.calendar_period != "CUSTOM" ? var.calendar_period : null
```

Comment:
Lets move `var.calendar_period == "CUSTOM"` to a boolean local and we can reuse it here and below.